lib/web/ui/integration.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ui
+
+import (
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+// IntegrationAWSOIDCSpec contain the specific fields for the `aws-oidc` subkind integration.
+type IntegrationAWSOIDCSpec struct {
+	// RoleARN is the role associated with the integration when SubKind is `aws-oidc`
+	RoleARN string `json:"roleArn,omitempty"`
+}
+
+// Integration describes the base Integration fields (can be read or wri) Integration
+type Integration struct {
+	// Name is the Integration name.
+	Name string `json:"name"`
+	// SubKind is the Integration SubKind.
+	SubKind string `json:"subKind"`
+	// AWSOIDC contains the fields for `aws-oidc` subkind integration.
+	AWSOIDC *IntegrationAWSOIDCSpec `json:"awsoidc"`
+}
```

Comment:
I know this is a draft and probably more work coming but wanted to double check what i can expect for response (the json naming on below example doesn't have to match, i just stubbed things until the real implementation came). Below is the web screen for listing integrations and plugins:
<img width="963" alt="image" src="https://user-images.githubusercontent.com/43280172/230457713-107119ff-b61d-4bac-b173-8d2a9297f3aa.png">

i built this object based on existing `plugin` object: https://github.com/gravitational/teleport/pull/24108/files#diff-39d6ead4c09146a78b2dd1fb13ab281b908f85034c6de9eaddfcef6cd53a7ecfR33

```ts
function makeIntegration(json: any): Integration {
  json = json || {};
  const { name, details, status, type } = json; // this is the api response, ignore extra fields below, those are purely for UI use.
  return {
    resourceType: 'integration',
    name,
    details, // missing details? or will this be always be empty?
    statusDescription: status?.description,
    kind: type, // used to determine the icon, eg: kind `aws` will render aws icon, for integration this would be equivalent to `subKind` right?
    statusCode: status?.code, // missing status?
    statusCodeText: convertIntegrationCodeToText(status?.code),
  };
}

function convertIntegrationCodeToText(code = 0) {
  if (code === IntegrationCode.Running) {
    return 'Running';
  }
  if (code === IntegrationCode.Paused) {
    return 'Paused';
  }
  if (code === IntegrationCode.Error) {
    return 'Error';
  }
  return 'Unspecified';
}
```

Reply:
> // missing details? or will this be always be empty?
There's no details so far
We can add them, show nothing or add a static value

// used to determine the icon, eg: kind `aws` will render aws icon, for integration this would be equivalent to `subKind` right?
Yes, this should read from subkind. The only value for now is `aws-oidc`

Reply:
> We actually removed the status field #24121 Sorry

ah gotcha, so if i understand correctly... it will always be in the `running` state correct?

Reply:
Yes